### DISCUSSION TITLE
Can I use React Query in Next JS for client-side fetching (substitute for SWR)?

### DISCUSSION THREAD

**Discussion by florianmartens** [TanStack/query#2308] [Q&A]
I'm currently building a project in NextJS. Next to some sites that will be built via SSR, I have a lot of Dashboard functionality that does not need any SEO or static capabilities. In these cases, Next recommends a library called SWR but I'd much rather use React Query for all my client-side fetching.

I wasn't able to find much information on this, but my gut feeling says that this should work just as well as using React Query with plain React.

However, I wanted to be sure that I'm not running into any unforeseen pitfalls here as I'm quite new to Next.

**Can I use React Query in Next JS for client-side fetching (substitute for SWR)?**

**Comment by TkDodo** [comment]
yes, see:

- https://react-query.tanstack.com/guides/ssr#using-nextjs
- https://react-query.tanstack.com/examples/nextjs
